scripts/cleanup.sh

```diff
@@ -3,26 +3,177 @@ set -ex
+    echo "Deleting {{workingDirectory}}/TOE_*"
+    sudo rm -rf {{workingDirectory}}/TOE_*
+fi
+if [[ $(sudo find {{workingDirectory}}/TOE_* -type d | sudo wc -l) -gt 0 ]]; then
+    echo "Failed to delete {{workingDirectory}}/TOE_*"
+    exit 1
+fi
+
+echo "Cleaning up ssm log files"
+if [[ $(sudo find /var/log/amazon/ssm -type f | sudo wc -l) -gt 0 ]]; then
+    echo "Deleting files within /var/log/amazon/ssm/*"
+    sudo find /var/log/amazon/ssm -type f -exec shred -zuf {} \;
+fi
+if [[ $(sudo find /var/log/amazon/ssm -type f | sudo wc -l) -gt 0 ]]; then
+    echo "Failed to delete /var/log/amazon/ssm"
+    exit 1
+fi
+if [[ -d "/var/log/amazon/ssm" ]]; then
+    echo "Deleting /var/log/amazon/ssm/*"
+    sudo rm -rf /var/log/amazon/ssm
+fi
+if [[ -d "/var/log/amazon/ssm" ]]; then
+    echo "Failed to delete /var/log/amazon/ssm"
+    exit 1
+fi
+
+if [[ $(sudo find /var/log/sa/sa* -type f | sudo wc -l) -gt 0 ]]; then
+    echo "Deleting /var/log/sa/sa*"
+    sudo shred -zuf /var/log/sa/sa*
+fi
+if [[ $(sudo find /var/log/sa/sa* -type f | sudo wc -l) -gt 0 ]]; then
+    echo "Failed to delete /var/log/sa/sa*"
+    exit 1
+fi
+
+if [[ $(sudo find /var/lib/dhclient/dhclient*.lease -type f | sudo wc -l) -gt 0 ]]; then
+    echo "Deleting /var/lib/dhclient/dhclient*.lease"
+    sudo shred -zuf /var/lib/dhclient/dhclient*.lease
+fi
+if [[ $(sudo find /var/lib/dhclient/dhclient*.lease -type f | sudo wc -l) -gt 0 ]]; then
+    echo "Failed to delete /var/lib/dhclient/dhclient*.lease"
+    exit 1
+fi
+
+if [[ $(sudo find /var/tmp -type f | sudo wc -l) -gt 0 ]]; then
+    echo "Deleting files within /var/tmp/*"
+    sudo find /var/tmp -type f -exec shred -zuf {} \;
+fi
+if [[ $(sudo find /var/tmp -type f | sudo wc -l) -gt 0 ]]; then
+    echo "Failed to delete /var/tmp"
+    exit 1
+fi
+if [[ $(sudo ls /var/tmp | sudo wc -l) -gt 0 ]]; then
+    echo "Deleting /var/tmp/*"
+    sudo rm -rf /var/tmp/*
+fi
+
+# Shredding is not guaranteed to work well on rolling logs
+
+if [[ -f "/var/lib/rsyslog/imjournal.state" ]]; then
+    echo "Deleting /var/lib/rsyslog/imjournal.state"
+    sudo shred -zuf /var/lib/rsyslog/imjournal.state
+    sudo rm -f /var/lib/rsyslog/imjournal.state
+fi
+
+if [[ $(sudo ls /var/log/journal/ | sudo wc -l) -gt 0 ]]; then
+    echo "Deleting /var/log/journal/*"
+    sudo find /var/log/journal/ -type f -exec shred -zuf {} \;
+    sudo rm -rf /var/log/journal/*
+fi
+
+# delete a few items missed in https://docs.aws.amazon.com/imagebuilder/latest/userguide/security-best-practices.html
 sudo rm -rf \
-    /etc/hostname \
     /etc/machine-id \
-    /etc/resolv.conf \
-    /etc/ssh/ssh_host* \
     /home/ec2-user/.ssh/authorized_keys \
-    /root/.ssh/authorized_keys \
-    /var/lib/cloud/data \
-    /var/lib/cloud/instance \
-    /var/lib/cloud/instances \
-    /var/lib/cloud/sem \
-    /var/lib/dhclient/* \
+    /var/cache/yum \
+    /tmp/* \
     /var/lib/dhcp/dhclient.* \
```

Comment:
Curious how we know those files ought to be deleted and would not cause any side effects?

Reply:
most of this was originally just taken directly from the amazon-eks-ami cleanup process: https://github.com/awslabs/amazon-eks-ami/blob/master/scripts/install-worker.sh#L343-L361